plugins/plugin-chart-echarts/src/Pie/transformProps.ts

```diff
@@ -71,6 +74,7 @@ export function formatPieLabel({
 export default function transformProps(chartProps: ChartProps): EchartsProps {
   const { width, height, formData, queriesData } = chartProps;
   const data: DataRecord[] = queriesData[0].data || [];
+  const coltypes: Array<GenericDataType> = queriesData[0].coltypes || [];
```

Comment:
`coltypes` should have already been specified in `ChartDataResponseResult`. You just need to extend the generic `ChartProps` with ECharts specific props like in table chart:

https://github.com/apache-superset/superset-ui/blob/22f2fc1dcfa1653300b5588fc7265dbb9a80e4e4/plugins/plugin-chart-table/src/types.ts#L72